TypeSystem/Reflection/test.cc

```diff
@@ -193,8 +232,55 @@ TEST(Reflection, VisitAllFields) {
       "255,65535,4294967295,18446744073709551615,"
       "-128,-32768,-2147483648,-9223372036854775808,"
       "1e+38,1e+308,"
-      "The String",
+      "The String,"
+      "Minus eight point five:-9.500000,"
+      "[-1,-2,-4],"
+      "[key1:value1,key2:value2]",
       bricks::strings::Join(result, ','));
 }
 
+namespace reflection_test {
+
+CURRENT_STRUCT(X) { CURRENT_FIELD(i, int32_t); };
+CURRENT_STRUCT(Y) { CURRENT_FIELD(v, std::vector<X>); };
+CURRENT_STRUCT(Z, Y) {
+  CURRENT_FIELD(d, double);
+  CURRENT_FIELD(v2, std::vector<std::vector<Y>>);
+};
+}
```

Comment:
Note: Need to test multiple dependency on the same struct too. Ex., if `X` contains `T`, and `Z` contains `T` as well.